### PR TITLE
Add Rectangle factory functions

### PR DESCRIPTION
**Pull request by DanRStevens**
Based on PR #352 (Add `Vector`). (Only the last commit is new).

Adds factory methods to create a `Rectangle` from either a `Point` and a `Vector`, or from two `Point`s.

### CODE REVIEW

File: include/NAS2D/Renderer/Rectangle.h [hunk 1]
```diff
@@ -30,6 +31,26 @@ struct Rectangle
 		mH(height)
 	{}
 
+	// Factory method
+	static Rectangle<BaseType> Create(Point<BaseType> startPoint, Vector<BaseType> size) {
```
Comment (cugone) [inline include/NAS2D/Renderer/Rectangle.h]: `startPoint` is ambiguous. Is it a corner? The center? Why not make it a `Vector`?

Comment (DanRStevens) [reply include/NAS2D/Renderer/Rectangle.h]: A `Rectangle` would not be well defined by two `Vector` objects. It needs either two corner `Point` objects, or it  needs one corner `Point` object, and a `Vector` for the size. The other corner point is then `startPoint + vector`.

And yes, `startPoint` is a corner, which has the minimum `(x, y)` value. Wasn't sure what to call it.

Comment (cugone) [reply include/NAS2D/Renderer/Rectangle.h]: I suggest renaming `startPoint` to remove external ambiguity.

Comment (DanRStevens) [reply include/NAS2D/Renderer/Rectangle.h]: Yes, but do you have a suggestion?

Comment (cugone) [reply include/NAS2D/Renderer/Rectangle.h]: The easiest is to name is `center`, `size` should be `radius` or `halfExtents` and then internally derive the resulting `Rectangle`:

```
static Rectangle<BaseType> Create(Point<BaseType> center, Vector<BaseType> halfExtents) {
    return {
        center.x - halfExtents.x(),
        center.y - halfExtents.y(), //Assuming topleft origin.
        halfExtents.x() * static_cast<BaseType>(2.0),
        halfExtents.y() * static_cast<BaseType>(2.0),
    };
}
```

Comment (DanRStevens) [reply include/NAS2D/Renderer/Rectangle.h]: Well, I wouldn't mind having additional overloads, such as being able to create from a center point. That is something I would like to support eventually.

In this case though, the intention is to create from a corner point and total size along each side. There are many use cases where a corner point is easily known, but a center point would need to be calculated. Plus the corner point and size representation closely matches the internal representation, so if that's what's already available, there should be a quick way to create a `Rectangle` from that information, without needing to convert to some intermediate representation and then back again.

----

As for the parameter names `startPoint` and `size`, they do have a certain symmetry to the other factory method which uses the names `startPoint` and `endPoint`.

----

I would advise against using `radius` for something that isn't a circle. I find that pretty confusing. My first thought there is either a square inscribed in a circle, or a square which forms the bounding box of a circle.

Comment (cugone) [reply include/NAS2D/Renderer/Rectangle.h]: Looking back on it, `startPoint` and `endPoint` are just synonyms for `mins` and `maxs`. I'm okay with this. Just not the internal representation as "corner and size". 😝

**Pull request merged**